Sources/TuistServer/Services/AuthTokenRefreshService.swift

```diff
@@ -0,0 +1,81 @@
+import FileSystem
+import Foundation
+import Mockable
+import Path
+import ServiceContextModule
+import TuistCore
+import TuistSupport
+
+@Mockable
+public protocol AuthTokenRefreshServicing {
+    func refreshTokens(
+        path: AbsolutePath,
+        serverURL: URL
+    ) async throws
+}
+
+public struct AuthTokenRefreshService: AuthTokenRefreshServicing {
+    private let fileSystem: FileSysteming
+    private let refreshAuthTokenService: RefreshAuthTokenServicing
+    private let serverCredentialsStore: ServerCredentialsStoring
+    private let serverSessionController: ServerSessionControlling
+    private let serverAuthenticationController: ServerAuthenticationControlling
+
+    // MARK: - Init
+
+    public init(
+        fileSystem: FileSysteming = FileSystem(),
+        refreshAuthTokenService: RefreshAuthTokenServicing = RefreshAuthTokenService(),
+        serverCredentialsStore: ServerCredentialsStoring = ServerCredentialsStore(),
+        serverSessionController: ServerSessionControlling = ServerSessionController(),
+        serverAuthenticationController: ServerAuthenticationControlling = ServerAuthenticationController()
+    ) {
+        self.fileSystem = fileSystem
+        self.refreshAuthTokenService = refreshAuthTokenService
+        self.serverCredentialsStore = serverCredentialsStore
+        self.serverSessionController = serverSessionController
+        self.serverAuthenticationController = serverAuthenticationController
+    }
+
+    public func refreshTokens(
+        path _: AbsolutePath,
+        serverURL: URL
+    ) async throws {
+        guard let token = try await serverAuthenticationController.authenticationToken(serverURL: serverURL)
+        else {
+            throw ServerClientAuthenticationError.notAuthenticated
+        }
+
+        switch token {
+        case let .user(legacyToken: _, accessToken: _, refreshToken: refreshToken):
+            if let refreshToken {
+                try await fetchTokens(serverURL: serverURL, refreshToken: refreshToken.token)
+            }
+        case _:
```

Comment:
```suggestion
        default:
```

The convention for a default case is to use the `default` keyword instead.

However, the rule of thumb is to be exhaustive unless you have a good reason to, so I'd change this to:
```suggestion
        case .project:
          break
```